src/tests/ftest/util/server_utils_base.py

```diff
@@ -146,6 +147,20 @@ def update_pattern(self, mode, host_qty):
             self.pattern = self.NORMAL_PATTERN
         self.pattern_count = host_qty * len(self.yaml.engine_params)
 
+    def update_pattern_timeout(self):
```

Comment:
Super nit - 1. "update" commonly implies that this function will accept some argument from the caller to update something. 2. it also only updates if not already set. I wonder if something like `update_default_pattern_timeout`, `set_default_pattern_timeout`, etc. might be more clear.